lib/DevServerManager.ts

```diff
@@ -59,21 +97,37 @@ class DevServerManager {
     }
   }
 
-  arrangeComponentsByType(components) {
+  arrangeComponentsByType(components: Component[]): ComponentsByType {
     return components.reduce((acc, component) => {
       if (!acc[component.type]) {
         acc[component.type] = {};
       }
 
-      acc[component.type][component.config.name] = component;
+      if ('name' in component.config && component.config.name) {
+        acc[component.type][component.config.name] = component;
+      }
 
       return acc;
-    }, {});
+    }, {} as ComponentsByType);
```

Comment:
If you make the change to `ComponentsByType` above, you can do this without casting

```
  arrangeComponentsByType(components: Component[]): ComponentsByType {
    return components.reduce<ComponentsByType>((acc, component) => {
      if (!acc[component.type]) {
        acc[component.type] = {};
      }

      if ('name' in component.config && component.config.name) {
        acc[component.type]![component.config.name] = component;
      }

      return acc;
    }, {});
  }
```

Reply:
done in [086af30](https://github.com/HubSpot/hubspot-cli/pull/1335/commits/086af30e65d2a8ef88ded01a818d1388441170dd)